presto-main/src/main/java/com/facebook/presto/util/PrestoIterators.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+import com.google.common.collect.AbstractIterator;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Iterator;
+
+public class PrestoIterators
+{
+    public static <T> Iterator<T> asCloseable(Iterator<T> iterator, Closeable resource)
```

Comment:
The concept is good, but the naming isn't. I'd expect `asCloseable(..)` to return a `Closeable` instance, which it doesn't do. What about

```
Iterator<T> closeWhenExhausted(Iterator<T> source, AutoCloseable closer)
```
?